src/main/java/org/sunbird/workflow/service/impl/UserProfileWfServiceImpl.java

```diff
@@ -133,13 +246,60 @@ public HashMap<String, Object> getUsersResult(Set<String> userIds) {
 				}
 			}
 		} catch (Exception e) {
-			logger.error(e);
+			logger.error("Exception while fetching user setails : ",e);
 			throw new ApplicationException("Hub Service ERROR: ", e);
 		}
 		return userResult;
 	}
 
-	private Map<String, Object> getSearchObject(Set<String> userIds) {
+	private Object userProfileRead(String userId) {
+		StringBuilder builder = new StringBuilder(configuration.getLmsServiceHost());
+		String endPoint = configuration.getUserProfileReadEndPoint().replace(Constants.USER_ID_VALUE, userId);
+		builder.append(endPoint);
+		Object response = requestServiceImpl.fetchResultUsingGet(builder);
+		Map<String, Object> readResponse = mapper.convertValue(response, Map.class);
+		return readResponse;
+	}
+
+	private Object migrateUser(WfRequest wfRequest) {
+
+		Map<String, Object> migrateRequestWrapper = new HashMap<>();
+		Map<String, Object> migrateRequestObject = new HashMap<>();
+		migrateRequestWrapper.put(Constants.USER_ID, wfRequest.getApplicationId());
+		migrateRequestWrapper.put(Constants.CHANNEL, wfRequest.getDeptName());
+		migrateRequestWrapper.put(Constants.FORCE_MIGRATION, true);
+		migrateRequestWrapper.put(Constants.SOFT_DELETE_OLD_GROUP, true);
+		migrateRequestWrapper.put(Constants.NOTIFY_MIGRATION, false);
+		migrateRequestObject.put(Constants.REQUEST, migrateRequestWrapper);
+		StringBuilder builder = new StringBuilder(configuration.getLmsServiceHost());
+		builder.append(configuration.getUserProfileMigrateEndPoint());
+		Map<String, Object> migrateUserApiResp = (Map<String, Object>) requestServiceImpl
+				.fetchResultUsingPatch(builder, migrateRequestObject, Map.class, getHeaders());
+		return migrateUserApiResp;
+	}
+
+	private void failedCase(WfRequest wfRequest) {
+		wfRequest.setState(Constants.FAILED);
+		wfRequest.setAction(Constants.FAILED);
+		workflowService.workflowTransition(Constants.ROOT_ORG, Constants.ORG, wfRequest, wfRequest.getApplicationId());
```

Comment:
is this correct behaviour ?
The idea is just to update the DB that this particular wfRequest is failed...